DXMainClient/DXGUI/Generic/OptionPanels/DisplayOptionsPanel.cs

```diff
@@ -787,6 +788,13 @@ public override bool Save()
 
             IniSettings.Renderer.Value = selectedRenderer.InternalName;
 
+#if ISWINDOWS
+            if (selectedRenderer.InternalName == "DDRAWCompat")
```

Comment:
Actually, this fix is made for any ddraws (as long as the selected renderer is not "no ddraw") -- a desync issue was confirmed in a machine with cnc-ddraw and win7 compatibility.

Besides, I'd suggest moving this if-statement into the `CheckAndPromptFix` function, since if the user intends not to use ddraw, he/she might needs the compatibility registry value (and therefore the same determination needs to be performed in `Startup`).